core/io/packet_peer.cpp

```diff
@@ -281,7 +281,7 @@ void PacketPeerStream::set_input_buffer_max_size(int p_max_size) {
 
 	//warning may lose packets
 	ERR_FAIL_COND_MSG(ring_buffer.data_left(), "Buffer in use, resizing would cause loss of data.");
-	ring_buffer.resize(nearest_shift(p_max_size + 4));
+	ring_buffer.resize(nearest_shift((p_max_size < 0 ? 0 : p_max_size) + 4));
```

Comment:
I don't think negative `p_max_size` is meant to be supported. So it should be an `ERR_FAIL_COND` IMO, not a hack at the end location where the warning is raised. @Faless